packages/typespec-client-generator-core/src/cache.ts

```diff
@@ -382,7 +493,10 @@ function buildHierarchyOfOperationGroups(
   return undefined;
 }
 
-function isArm(service: Namespace): boolean {
+function isArm(service: Namespace[] | Namespace): boolean {
+  if (Array.isArray(service)) {
```

Comment:
I go through the code,  the function parameter of service would `Namespace`  or  `Namespace[] | Namespace`.
And we use different logic to handle 'service' throughout the code.  I suggest to make it consistent at the early stage.